src/helpers/host-component-names.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Text, TextInput, View } from 'react-native';
+import TestRenderer from 'react-test-renderer';
+import { configureInternal, getConfig } from '../config';
+import { getQueriesForElement } from '../within';
+
+const defaultErrorMessage = `There seems to be an issue with your configuration that prevents the library from working correctly.
+Please ensure that you have a version of react native compatible with the library`;
+
+export function detectHostComponentNamesIfNeeded(): void {
+  if (getConfig().hostComponentNames) {
+    return;
+  }
+
+  try {
+    const renderer = TestRenderer.create(
+      <View>
+        <Text testID="text">Hello</Text>
+        <TextInput testID="textInput" />
+      </View>
+    );
+    const { getByTestId } = getQueriesForElement(renderer.root);
+    const textHostName = getByTestId('text').type;
+    const textInputHostName = getByTestId('textInput').type;
+
+    if (
+      typeof textHostName !== 'string' ||
+      typeof textInputHostName !== 'string'
+    ) {
+      throw new Error(defaultErrorMessage);
```

Comment:
I wonder if we can actually cover this code path. In order to do so, we would have to make `getByTestId` return non-host component. Therefore, seems like we could remove these check from here. @pierrezimmermannbam wdyt?

Reply:
This path shouldn't happen because getByTestId will always return host elements but for now typescript isn't aware of that. It's not easy to change the queries typing right now because some still return composite components so what I had in mind was to update the queries return type to HostReactInstance once we prepare v12 and remove legacy code, does that seem reasonable to you ?

Reply:
Let's just add short comment saying that this code like
```
// this code path should not happen as getByTestId should always return a host element
```